### PR TITLE
feat(TripPlanner): Use Stop ID to query OTP

### PR DESCRIPTION
**Pull request by kotva006**
#### Summary of changes
This pull request allows us to query OTP using stop ids.  This is supposed to remove walking directions to get to the initial stop if the user is already assumed at the stop.

**Asana Ticket:** [Fix Trip planner origin and destinations](https://app.asana.com/0/home/1201123880594717/1205711841709226)

---

#### General checks
* [ ] **Are the changes organized into self-contained commits with descriptive and well-formatted commit messages?** This is a good practice that can facilitate easier reviews.
* [ ] **Testing.** Do the changes include relevant passing updates to tests? This includes updating screenshots. Preferably tests are run locally to verify that there are no test failures created by these changes, before opening a PR.
* [ ] **Tech debt.** Have you checked for tech debt you can address in the area you're working in? This can be a good time to address small issues, or create Asana tickets for larger issues.

#### New UI, or substantial UI changes
* [ ] **Cross-browser compatibility** is less of an issue now that we're no longer supporting IE, but changes still need to work as expected in Safari, Chrome, and Firefox.
* [ ] **Are interactive elements accessible?** This includes at minimum having [relevant keyboard interactions](https://thinkdobecreate.com/articles/4-required-tests-before-shipping-new-features/#keyboard-interaction) and [visible focus](https://thinkdobecreate.com/articles/4-required-tests-before-shipping-new-features/#visible-focus), but can also include verification with screen reader testing.
* [ ] **Other accessibility checks** such as sufficient [color constrast](https://thinkdobecreate.com/articles/4-required-tests-before-shipping-new-features/#color-contrast), or whether the layout holds up at 200% [zoom level](https://thinkdobecreate.com/articles/4-required-tests-before-shipping-new-features/#visible-focus).

#### New endpoints, or non-trivial changes to current endpoints
* [ ] **Have we load-tested** any new pages or internal API endpoints that will receive significant traffic? [See load testing docs](./../apps/site/load_tests/README.md)
* [ ] **If this change involves routes,** does it work correctly with pertinent "unusual" routes such as the combined Green Line, Silver Line, Foxboro commuter rail, and single-direction bus routes like the 170?

### CODE REVIEW

**Review by anthonyshull — APPROVED** [review]
Overall looks good. Just a few questions.

File: apps/site/assets/js/trip-planner-location-controls.js [hunk 1]
```diff
@@ -357,6 +363,13 @@ export class TripPlannerLocControls {
     };
   }
 
+  setStopValue(ac, hit) {
+    if (hit.stop?.id) {
+      const stopIdEl = this.getById(ac._selectors.stop_id);
+      stopIdEl.value = hit.stop.id;
```
Comment (anthonyshull) [inline apps/site/assets/js/trip-planner-location-controls.js:369]: Is there a reason to set a const here?

```
this.getById(ac._selectors.stop_id).value = hit.stop.id;
```

**Comment by anthonyshull** [comment]
Just so I know, are there no tests to write here?

File: apps/site/assets/js/trip-planner-location-controls.js (hunk 1 above)
Comment (kotva006) [reply apps/site/assets/js/trip-planner-location-controls.js:369]: No reason, just preference

**Comment by thecristen** [comment]
I had to revert this in release 2023.11.22.02 as it wasn't working well when the origin and destination were _not_ stop IDs.

It raises a GraphQL exception when I try to plan a trip between Fall River Historical Society, 451 Rock St, Fall River, MA, 02720, USA and Worcester locally, here's the response the query returns:

```elixir
%{
    "data" => %{"plan" => nil},
    "errors" => [
        %{
            "extensions" => %{"classification" => "DataFetchingException"},
            "locations" => [%{"column" => 3, "line" => 2}],
            "message" => "Exception while fetching data (/plan) : Missing mandatory id on FeedScopeId [Value cannot be null, empty or just whitespace: '']",
            "path" => ["plan"]
        }
    ]
}
```

And when interacting with it on the front-end, I noticed if I edited the destination for a working trip plan (e.g. I've already pulled up itineraries) from a stop to an address, and regenerate the plan, the itineraries would appear to update, but are actually routing me to the initially selected stop. Not sure what's going on there.

**Pull request merged**